agents/sorenson_dlm/dlm_agent.py

```diff
@@ -0,0 +1,259 @@
+        self.dlm = DLM(ip_address, int(port))
+        agg_params = {'frame length': 60, }
+        self.agent.register_feed('voltages',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+
+    def start_acq(self, session, params=None):
+        '''
+        Get voltage and current values from the sorenson, publishes them to
+        the feed
+
+        Args:
+            sampling_frequency: defaults to 2.5Hz
+        '''
+        if params is None:
+            params = {}
+
+        f_sample = params.get('sampling_frequency')
+        if f_sample is None:
+            f_sample = self.f_sample
+
+        sleep_time = 1./f_sample - 0.01
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn("Could not start init because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'voltages',
+                    'data': {}
+                }
+                voltage_reading = self.dlm.read_voltage()
+                current_reading = self.dlm.read_current()
+                print('Voltage: {}'.format(voltage_reading))
+                print('Current: {}'.format(current_reading))
+                data['data']["voltage"] = np.float(voltage_reading)
+                data['data']["current"] = np.float(current_reading)
+
+                self.agent.publish_to_feed('voltages', data)
+                time.sleep(sleep_time)
+
+            self.agent.feeds['voltages'].flush_buffer()
+        return True, 'Acquistion exited cleanly'
+
+    def set_voltage(self, session, params=None):
+        """
+        Sets voltage of power supply:
+        Args:
+            voltage (float): Voltage to set.
+        
+        Examples:
+            Example of a client, setting the current to 1V:
+
+                client.set_voltage(voltage = 1.)
+
+        """
+
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            if acquired:
+                if self.over_volt == 0:
+                    return False, 'Over voltage protection not set'
+                elif float(params['voltage']) > float(self.over_volt):
+                    return False, 'Voltage greater then over voltage protection'
+                else:
+                    self.dlm.send_msg('SOUR:VOLT {}'.format(params['voltage']))
+            else:
+                return False, "Could not acquire lock"
+
+        return True, 'Set voltage to {}'.format(params['voltage'])
+
+    def set_over_volt(self, session, params=None):
+        """
```

Comment:
Done for most tasks, can I assume `start_acq` and `stop_acq` don't need this?

Reply:
`start_acq` does, `stop_acq` doesn't. A subtle thing with this method signature override is the function names need to match, and we want to present in the documentation the function name registered for use by a client. So actually `start_acq` should be renamed to `acq`. Also, since we don't want `stop_acq` to really show up at all in the docs, `stop_acq` should be renamed to `_stop_acq` as well.

This is done in many other agents at this point, so there are some examples, see for instance: https://github.com/simonsobs/socs/blob/4ce174e0c5dd512db992f0818d3722cdb9fb646e/agents/pfeiffer_tc400/pfeiffer_tc400_agent.py#L263

Reply:
There's some small formatting issues here, fixing in a commit that I'll push shortly.